File: packages/node/iii-example/src/state.ts
import { iii } from './iii'
import type {
  IState,
  StateDeleteInput,
  StateDeleteResult,
  StateGetInput,
  StateListInput,
  StateSetInput,
  StateSetResult,
  StateUpdateInput,
  StateUpdateResult,
} from 'iii-sdk/state'

export const state: IState = {
  get: <TData>(input: StateGetInput): Promise<TData | null> => iii.call('state.get', input),
  set: <TData>(input: StateSetInput): Promise<StateSetResult<TData> | null> =>
    iii.call('state.set', input),
  delete: (input: StateDeleteInput): Promise<StateDeleteResult> => iii.call('state.delete', input),
  list: <TData>(input: StateListInput): Promise<TData[]> => iii.call('state.list', input),
  update: <TData>(input: StateUpdateInput): Promise<StateUpdateResult<TData> | null> =>
    iii.call('state.update', input),
}

File: packages/python/iii-example/src/state.py
from typing import Any

from .iii import get_iii


class State:
    async def get(self, scope: str, key: str) -> Any | None:
        iii = get_iii()
        return await iii.call("state::get", {"scope": scope, "key": key})

    async def set(self, scope: str, key: str, data: Any) -> Any:
        iii = get_iii()
        return await iii.call("state::set", {"scope": scope, "key": key, "data": data})

    async def delete(self, scope: str, key: str) -> None:
        iii = get_iii()
        return await iii.call("state::delete", {"scope": scope, "key": key})

    async def get_group(self, scope: str) -> list[Any]:
        iii = get_iii()
        return await iii.call("state::list", {"scope": scope})


state = State()
